File: frontend/src/api/onboarding.ts
import request from '@/utils/request'
import type { ApiResponse } from '@/types'

// ── A5 OnboardingRunRequest（对齐 backend/app/schemas/onboarding.py）──

export interface EnterpriseInfo {
  name: string
  industry?: string
  license_no?: string
  contact_name?: string
  contact_phone?: string
}

export interface BrandInfo {
  name: string
  differentiator?: string
  slogan?: string
}

export interface StoreInfo {
  name: string
  city?: string
  district?: string
  address?: string
  phone?: string
  business_hours?: string
  is_primary?: boolean
}

export interface ServiceInfo {
  name: string
  description?: string
  category?: string
  price_hint?: string
  duration_minutes?: number
}

export interface SeedFact {
  title: string
  content: string
}

export interface OnboardingRunRequest {
  enterprise: EnterpriseInfo
  brand?: BrandInfo
  stores: StoreInfo[]
  services: ServiceInfo[]
  competitors: string[]
  target_customers: string
  raw_inputs: string
  seed_facts: SeedFact[]
  target_engines: string[]
  search_enabled: boolean
}

export interface OnboardingTaskResponse {
  task_id: number
  status: string
  progress_pct: number
  progress_message: string
  step: string
  result?: any
}

// ── API 函数 ──

export function runOnboarding(data: OnboardingRunRequest) {
  return request.post<any, ApiResponse<OnboardingTaskResponse>>('/api/v1/user/onboarding/run', data)
}

/** 创建 SSE 连接监听入驻进度。返回 EventSource，调用方负责 close。 */
export function subscribeOnboardingEvents(taskId: number): EventSource {
  const base = import.meta.env.VITE_API_BASE || import.meta.env.VITE_API_BASE_URL || ''
  const token = localStorage.getItem('geo-user')
    ? JSON.parse(localStorage.getItem('geo-user') || '{}').token || ''
    : ''
  // SSE 需要 token 鉴权，通过 query param 传递（FastAPI SSE 不支持 header）
  const url = `${base}/api/v1/user/onboarding/events/${taskId}?token=${encodeURIComponent(token)}`
  return new EventSource(url)
}

File: backend/app/schemas/onboarding.py
"""入驻向导 — OnboardingRunRequest schema。"""
from typing import List, Optional

from pydantic import Field

from app.schemas.common import BaseSchema


class EnterpriseInfo(BaseSchema):
    """企业基本信息（Step 0 表单）。"""
    name: str = Field(..., min_length=1, max_length=200)
    industry: str = "beauty_local"
    license_no: Optional[str] = None
    contact_name: Optional[str] = None
    contact_phone: Optional[str] = None


class BrandInfo(BaseSchema):
    """品牌信息。"""
    name: str = Field(..., min_length=1, max_length=200)
    differentiator: Optional[str] = None
    slogan: Optional[str] = None


class StoreInfo(BaseSchema):
    """门店信息。"""
    name: str = Field(..., min_length=1, max_length=200)
    city: Optional[str] = None
    district: Optional[str] = None
    address: Optional[str] = None
    phone: Optional[str] = None
    business_hours: Optional[str] = None
    is_primary: bool = True


class ServiceInfo(BaseSchema):
    """服务项目。"""
    name: str = Field(..., min_length=1, max_length=200)
    description: Optional[str] = None
    category: Optional[str] = None
    price_hint: Optional[str] = None
    duration_minutes: Optional[int] = None


class SeedFact(BaseSchema):
    """种子 Fact。"""
    title: str = Field(..., min_length=1, max_length=500)
    content: str = Field(..., min_length=1)


class OnboardingRunRequest(BaseSchema):
    """POST /user/onboarding/run 请求体 — 一次性提交全部入驻数据。"""
    enterprise: EnterpriseInfo
    brand: Optional[BrandInfo] = None
    stores: List[StoreInfo] = Field(default_factory=list)
    services: List[ServiceInfo] = Field(default_factory=list)
    competitors: List[str] = Field(default_factory=list)
    target_customers: str = ""
    raw_inputs: str = ""
    seed_facts: List[SeedFact] = Field(default_factory=list)
    target_engines: List[str] = Field(default_factory=list)
    search_enabled: bool = True
